src/driver/amdxdna/aie2_debugfs.c

```diff
@@ -214,19 +214,21 @@ static int aie2_state_show(struct seq_file *m, void *unused)
 AIE2_DBGFS_FOPS(state, aie2_state_show, aie2_state_write);
 
 static ssize_t aie2_dpm_level_set(struct file *file, const char __user *ptr,
-				size_t len, loff_t *off)
+				  size_t len, loff_t *off)
 {
 	struct amdxdna_dev_hdl *ndev = file_to_ndev_rw(file);
 	u32 val;
 	int ret;
 
-	ret = kstrtouint_from_user(ptr, len, 10, &val);
+	ret = kstrtoint_from_user(ptr, len, 10, &val);
 	if (ret) {
 		XDNA_ERR(ndev->xdna, "Invalid input value: %d", val);
 		return ret;
 	}
 
-	ret = aie2_smu_set_dpm_level(ndev, val);
+	mutex_lock(&ndev->xdna->dev_lock);
+	ret = aie2_smu_set_fixed_dpm_level(ndev, val);
```

Comment:
So this "aie2_smu_set_fixed_dpm_level" called in the debugfs. The expectation is that, if we set certain dpm level through debugfs, do we want that to be constant irrespective of the session dpm level that was computed by the resolver? 
What happens to the session that wanted a different dpm level?

Reply:
Any time fixed dpm level is set (through debug fs or performance mode), resource solver is not allowed to change the dpm level. All sessions will use the fixed one. This is for debug purposes only. When fixed dpm level is not set, the default level will be used which is under control by resource solver.